pkg/openstack/instancesv2.go

```diff
@@ -151,6 +154,22 @@ func (i *InstancesV2) makeInstanceID(srv *servers.Server) string {
 
 func (i *InstancesV2) getInstance(ctx context.Context, node *v1.Node) (*ServerAttributesExt, error) {
 	if node.Spec.ProviderID == "" {
+		instanceID, err := i.metadataServer.GetInstanceID()
```

Comment:
in our case OCCM is executed in control-planes. If we execute this in control-plane it will return the current instance metadata and instance id. NOT for instance the node x which can be parameter for this function. As I see it, this will not work like this